[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-animate

Disallows the [`.animate`](https://api.jquery.com/animate/)/[`.stop`](https://api.jquery.com/stop/)/[`.finish`](https://api.jquery.com/finish/) methods. Use the `allowScroll` option to allow animations which are just used for scrolling. Prefer CSS transitions.

üìã This rule is enabled in `plugin:no-jquery/slim`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).animate();
$( 'div' ).stop();
$( 'div' ).finish();
$div.animate();
$( 'div' ).first().animate();
$( 'div' ).append( $( 'input' ).animate() );
$div.animate( { scrollTop: 100 } );
$div.animate( { scrollLeft: 200 } );
$div.animate( { scrollTop: 100, scrollLeft: 200 } );
$div.animate( { scrollTop: 100, width: 300 } );
```

‚úîÔ∏è Examples of **correct** code:
```js
animate();
[].animate();
div.animate();
div.animate;
stop();
[].stop();
div.stop();
div.stop;
finish();
[].finish();
div.finish();
div.finish;
```

‚ùå Examples of **incorrect** code with `[{"allowScroll":false}]` options:
```js
$div.animate( { scrollTop: 100 } );
```

‚ùå Examples of **incorrect** code with `[{"allowScroll":true}]` options:
```js
$div.animate();
$div.animate( { scrollTop: 100, width: 300 } );
```

‚úîÔ∏è Examples of **correct** code with `[{"allowScroll":true}]` options:
```js
$div.animate( { scrollTop: 100 } );
$div.animate( { scrollLeft: 200 } );
$div.animate( { scrollTop: 100, scrollLeft: 200 } );
$div.animate( { scrollTop: 100 } ).stop();
$div.animate( { scrollTop: 100 } ).finish();
```

## Resources

* [Rule source](/src/rules/no-animate.js)
* [Test source](/tests/rules/no-animate.js)
