[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-attr

Disallows the [`.attr`](https://api.jquery.com/attr/)/[`.removeAttr`](https://api.jquery.com/removeAttr/) methods and `$.attr`/[`$.removeAttr`](https://api.jquery.com/jQuery.removeAttr/) utilies. Prefer `Element#getAttribute`/`setAttribute`/`removeAttribute`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$.attr();
$( 'div' ).attr();
$div.attr();
$( 'div' ).first().attr();
$( 'div' ).append( $( 'input' ).attr() );
$( 'div' ).attr( 'name' );
$( 'div' ).attr( 'name', 'random' );
$.removeAttr();
$( 'div' ).removeAttr( 'name' );
```

‚úîÔ∏è Examples of **correct** code:
```js
attr();
[].attr();
div.attr();
div.attr;
removeAttr();
div.removeAttr;
```

## Resources

* [Rule source](/src/rules/no-attr.js)
* [Test source](/tests/rules/no-attr.js)
