[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-ajax

Disallows the [`$.ajax`](https://api.jquery.com/jQuery.ajax/)/[`$.get`](https://api.jquery.com/jQuery.get/)/[`$.getJSON`](https://api.jquery.com/jQuery.getJSON/)/[`$.getScript`](https://api.jquery.com/jQuery.getScript/)/[`$.post`](https://api.jquery.com/jQuery.post/) utilies. Prefer `Window.fetch`.

üìã This rule is enabled in `plugin:no-jquery/slim`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$.ajax();
$.get();
$.getJSON();
$.getScript();
$.post();
```

‚úîÔ∏è Examples of **correct** code:
```js
ajax();
div.ajax();
div.ajax;
get();
div.get();
div.get;
getJSON();
div.getJSON();
div.getJSON;
getScript();
div.getScript();
div.getScript;
post();
div.post();
div.post;
```

## Resources

* [Rule source](/src/rules/no-ajax.js)
* [Test source](/tests/rules/no-ajax.js)
