[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-browser

Disallows the [`$.browser`](https://api.jquery.com/jQuery.browser/) property.

üìã This rule is enabled in `plugin:no-jquery/deprecated-1.3`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$.browser;
$.browser.style;
```

‚úîÔ∏è Examples of **correct** code:
```js
browser;
a.browser;
browser.foo;
a.browser.foo;
$.browsers;
```

## Resources

* [Rule source](/src/rules/no-browser.js)
* [Test source](/tests/rules/no-browser.js)
