[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-closest

Disallows the [`.closest`](https://api.jquery.com/closest/) method. Prefer `Element#closest`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).closest();
$div.closest();
$( 'div' ).first().closest();
$( 'div' ).append( $( 'input' ).closest() );
```

‚úîÔ∏è Examples of **correct** code:
```js
closest();
[].closest();
div.closest();
div.closest;
```

## Resources

* [Rule source](/src/rules/no-closest.js)
* [Test source](/tests/rules/no-closest.js)
