[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-context-prop

Disallows the [`.context`](https://api.jquery.com/context/) property.

üìã This rule is enabled in `plugin:no-jquery/deprecated-1.10`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).context;
$div.context;
$div.context.prop;
$div.context.method();
$( 'div' ).first().context;
f( $div.context );
$( 'div' ).append( $( 'input' ).context );
```

‚úîÔ∏è Examples of **correct** code:
```js
context;
div.context;
$div.prop.context;
$div.context();
$div.context( arg );
```

## Resources

* [Rule source](/src/rules/no-context-prop.js)
* [Test source](/tests/rules/no-context-prop.js)
