[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-camel-case

Disallows the `$.camelCase` utility.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.camelCase();
```

âœ”ï¸ Examples of **correct** code:
```js
camelCase();
myClass.camelCase();
$div.camelCase();
```

## Resources

* [Rule source](/src/rules/no-camel-case.js)
* [Test source](/tests/rules/no-camel-case.js)
