[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-class

Disallows the [`.addClass`](https://api.jquery.com/addClass/)/[`.hasClass`](https://api.jquery.com/hasClass/)/[`.removeClass`](https://api.jquery.com/removeClass/)/[`.toggleClass`](https://api.jquery.com/toggleClass/) methods. User the `onlyDeprecated` option to only report deprecated usages. Prefer `Element#classList`.

üìã This rule is enabled in `plugin:no-jquery/deprecated-3.0` with `[{"onlyDeprecated":true}]` options.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).addClass();
$div.addClass();
$( 'div' ).first().addClass();
$( 'div' ).append( $( 'input' ).addClass() );
$( 'div' ).hasClass();
$div.hasClass();
$( 'div' ).first().hasClass();
$( 'div' ).append( $( 'input' ).hasClass() );
$( 'div' ).removeClass();
$div.removeClass();
$( 'div' ).first().removeClass();
$( 'div' ).append( $( 'input' ).removeClass() );
$( 'div' ).toggleClass();
$div.toggleClass();
$( 'div' ).first().toggleClass();
$( 'div' ).append( $( 'input' ).toggleClass() );
```

‚úîÔ∏è Examples of **correct** code:
```js
addClass();
[].addClass();
div.addClass();
div.addClass;
hasClass();
[].hasClass();
div.hasClass();
div.hasClass;
removeClass();
[].removeClass();
div.removeClass();
div.removeClass;
toggleClass();
[].toggleClass();
div.toggleClass();
div.toggleClass;
```

‚ùå Examples of **incorrect** code with `[{"onlyDeprecated":true}]` options:
```js
$div.toggleClass();
$div.toggleClass( false );
$div.toggleClass( true );
$div.toggleClass( undefined );
```

‚úîÔ∏è Examples of **correct** code with `[{"onlyDeprecated":true}]` options:
```js
$div.attr( 'class', '' );
$div.removeClass( 'myClass' );
toggleClass( false );
obj.toggleClass( false );
$div.toggleClass( 'myClass', true );
$div.toggleClass( 'myClass', false );
$div.toggleClass( 'myClass' );
$div.toggleClass( 'myClass', undefined );
```

## Resources

* [Rule source](/src/rules/no-class.js)
* [Test source](/tests/rules/no-class.js)
