[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-css

Disallows the [`.css`](https://api.jquery.com/css/) method and `$.css` utility. Prefer `Window#getComputedStyle`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.css();
$( 'div' ).css();
$div.css();
$( 'div' ).first().css();
$( 'div' ).append( $( 'input' ).css() );
```

âœ”ï¸ Examples of **correct** code:
```js
css();
[].css();
div.css();
div.css;
```

## Resources

* [Rule source](/src/rules/no-css.js)
* [Test source](/tests/rules/no-css.js)
