[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-done-fail

Disallows the [`.done`](https://api.jquery.com/deferred.done/)/[`.fail`](https://api.jquery.com/deferred.fail/) methods. Prefer `.then`.

📋 This rule is enabled in `plugin:no-jquery/all`.

## Rule details

❌ Examples of **incorrect** code:
```js
promise.done( callback );
promise.fail( callback );
```

✔️ Examples of **correct** code:
```js
promise.then( doneCallback, failCallback );
done();
fail();
```

## Resources

* [Rule source](/src/rules/no-done-fail.js)
* [Test source](/tests/rules/no-done-fail.js)
