[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-deferred

Disallows [`$.Deferred`](https://api.jquery.com/jQuery.Deferred/) constructor. Prefer `Promise`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.Deferred();
new $.Deferred();
```

âœ”ï¸ Examples of **correct** code:
```js
Deferred();
new Deferred();
'test'.Deferred();
'test'.Deferred;
```

## Resources

* [Rule source](/src/rules/no-deferred.js)
* [Test source](/tests/rules/no-deferred.js)
