[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-each-collection

Disallows the [`.each`](https://api.jquery.com/each/) method. Prefer `Array#forEach`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).each();
$div.each();
$( 'div' ).first().each();
$( 'div' ).append( $( 'input' ).each() );
```

‚úîÔ∏è Examples of **correct** code:
```js
each();
[].each();
div.each();
div.each;
$.each();
```

## Resources

* [Rule source](/src/rules/no-each-collection.js)
* [Test source](/tests/rules/no-each-collection.js)
