# no-error-shorthand

Disallows the [`.error`](https://api.jquery.com/error/) method. Prefer `.on` or `.trigger`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/deprecated-1.8`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$( 'div' ).error( handler );
$div.error();
$( 'div' ).first().error();
$( 'div' ).append( $( 'input' ).error() );
```

âœ”ï¸ Examples of **correct** code:
```js
error();
[].error();
div.error();
div.error;
$.error();
```

ğŸ”§ Examples of code **fixed** by using  `--fix`:
```js
$( 'div' ).error( handler );               /* â†’ */ $( 'div' ).on( 'error', handler );
$div.error();                              /* â†’ */ $div.trigger( 'error' );
$( 'div' ).first().error();                /* â†’ */ $( 'div' ).first().trigger( 'error' );
$( 'div' ).append( $( 'input' ).error() ); /* â†’ */ $( 'div' ).append( $( 'input' ).trigger( 'error' ) );
```

## Resources

* [Rule source](/src/rules/no-error-shorthand.js)
* [Test source](/src/tests/no-error-shorthand.js)
