[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-extend

Disallows the [`$.extend`](https://api.jquery.com/jQuery.extend/) utility. Prefer `Object.assign` or the spread operator.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$.extend( {}, foo );
$.extend( true, {}, foo );
```

‚úîÔ∏è Examples of **correct** code:
```js
extend();
myMethod.extend();
myMethod.extend;
```

‚ùå Examples of **incorrect** code with `[{"allowDeep":true}]` options:
```js
$.extend( {}, foo );
```

‚úîÔ∏è Examples of **correct** code with `[{"allowDeep":true}]` options:
```js
$.extend( true, {}, foo );
```

## Resources

* [Rule source](/src/rules/no-extend.js)
* [Test source](/tests/rules/no-extend.js)
