# no-error

Disallows the [`$.error`](https://api.jquery.com/jQuery.error/) utility. Prefer `throw`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.error( msg );
```

âœ”ï¸ Examples of **correct** code:
```js
error( msg );
myClass.error( msg );
$div.error( msg );
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
$.error( msg ); /* â†’ */ throw new Error( msg );
```

## Resources

* [Rule source](/src/rules/no-error.js)
* [Test source](/src/tests/no-error.js)
