[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-extend

Disallows the [`$.extend`](https://api.jquery.com/jQuery.extend/) utility. Prefer `Object.assign` or the spread operator. Use the `allowDeep` option to allow using the method with the `deep` argument.

üìã This rule is enabled in `plugin:no-jquery/all`.

üîß The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$.extend( {}, foo );
$.extend( true, {}, foo );
```

‚úîÔ∏è Examples of **correct** code:
```js
extend();
myMethod.extend();
myMethod.extend;
$.extend( { myUtil: fn } );
```

‚ùå Examples of **incorrect** code with `[{"allowDeep":true}]` options:
```js
$.extend( {}, foo );
$.extend( fooCouldBeNull, doesNotAutofix );
```

‚úîÔ∏è Examples of **correct** code with `[{"allowDeep":true}]` options:
```js
$.extend( true, {}, foo );
```

üîß Examples of code **fixed** by this rule:
```js
$.extend( {}, foo ); /* ‚Üí */ Object.assign( {}, foo );
```

üîß Examples of code **fixed** by this rule with `[{"allowDeep":true}]` options:
```js
$.extend( {}, foo ); /* ‚Üí */ Object.assign( {}, foo );
```

## Resources

* [Rule source](/src/rules/no-extend.js)
* [Test source](/tests/rules/no-extend.js)
