[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-fade

Disallows the [`.fadeIn`](https://api.jquery.com/fadeIn/)/[`.fadeOut`](https://api.jquery.com/fadeOut/)/[`.fadeTo`](https://api.jquery.com/fadeTo/)/[`.fadeToggle`](https://api.jquery.com/fadeToggle/) methods. Prefer CSS transitions.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/slim`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$( 'div' ).fadeIn();
$div.fadeIn();
$( 'div' ).first().fadeIn();
$( 'div' ).append( $( 'input' ).fadeIn() );
$( 'div' ).fadeOut();
$div.fadeOut();
$( 'div' ).first().fadeOut();
$( 'div' ).append( $( 'input' ).fadeOut() );
$( 'div' ).fadeTo();
$div.fadeTo();
$( 'div' ).first().fadeTo();
$( 'div' ).append( $( 'input' ).fadeTo() );
$( 'div' ).fadeToggle();
$div.fadeToggle();
$( 'div' ).first().fadeToggle();
$( 'div' ).append( $( 'input' ).fadeToggle() );
```

âœ”ï¸ Examples of **correct** code:
```js
fadeIn();
[].fadeIn();
div.fadeIn();
div.fadeIn;
fadeOut();
[].fadeOut();
div.fadeOut();
div.fadeOut;
fadeTo();
[].fadeTo();
div.fadeTo();
div.fadeTo;
fadeToggle();
[].fadeToggle();
div.fadeToggle();
div.fadeToggle;
```

## Resources

* [Rule source](/src/rules/no-fade.js)
* [Test source](/tests/rules/no-fade.js)
