[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-find-util

Disallows the `$.find` utility. Prefer `Document#querySelectorAll`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.find();
```

âœ”ï¸ Examples of **correct** code:
```js
find();
[].find();
div.find();
div.find;
$.extend().find();
$div.myPlugin( 'foo' ).find();
$( 'div' ).find();
$div.find();
$( 'div' ).first().find();
$( 'div' ).append( $( 'input' ).find() );
```

## Resources

* [Rule source](/src/rules/no-find-util.js)
* [Test source](/tests/rules/no-find-util.js)
