[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-filter

Disallows the [`.filter`](https://api.jquery.com/filter/) method and `$.filter` utility. Prefer `Array#filter`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.filter();
$( 'div' ).filter();
$div.filter();
$( 'div' ).first().filter();
$( 'div' ).append( $( 'input' ).filter() );
```

âœ”ï¸ Examples of **correct** code:
```js
filter();
[].filter();
div.filter();
div.filter;
```

## Resources

* [Rule source](/src/rules/no-filter.js)
* [Test source](/tests/rules/no-filter.js)
