[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-fx

Disallows `$.fx`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/slim`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.fx;
$.fx.interval;
$.fx.off;
$.fx.speeds.slow;
$.fx.start();
```

âœ”ï¸ Examples of **correct** code:
```js
fx;
fx.interval;
a.fx;
```

## Resources

* [Rule source](/src/rules/no-fx.js)
* [Test source](/tests/rules/no-fx.js)
