[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-grep

Disallows the [`$.grep`](https://api.jquery.com/jQuery.grep/) utility. Prefer `Array#filter`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.grep();
```

âœ”ï¸ Examples of **correct** code:
```js
grep();
'test'.grep();
'test'.grep;
```

## Resources

* [Rule source](/src/rules/no-grep.js)
* [Test source](/tests/rules/no-grep.js)
