[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-global-eval

Disallows the [`$.globalEval`](https://api.jquery.com/jQuery.globalEval/) utility.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.globalEval();
```

âœ”ï¸ Examples of **correct** code:
```js
globalEval();
'test'.globalEval();
'test'.globalEval;
```

## Resources

* [Rule source](/src/rules/no-global-eval.js)
* [Test source](/tests/rules/no-global-eval.js)
