[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-hold-ready

Disallows the [`$.holdReady`](https://api.jquery.com/jQuery.holdReady/) utility.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.2`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.holdReady();
```

âœ”ï¸ Examples of **correct** code:
```js
holdReady();
'test'.holdReady();
'test'.holdReady;
```

## Resources

* [Rule source](/src/rules/no-hold-ready.js)
* [Test source](/tests/rules/no-hold-ready.js)
