[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-is-empty-object

Disallows the [`$.isEmptyObject`](https://api.jquery.com/jQuery.isEmptyObject/) utility.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$.isEmptyObject();
```

‚úîÔ∏è Examples of **correct** code:
```js
isEmptyObject();
myClass.isEmptyObject();
$div.isEmptyObject();
```

## Resources

* [Rule source](/src/rules/no-is-empty-object.js)
* [Test source](/tests/rules/no-is-empty-object.js)
