[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-is-numeric

Disallows the [`$.isNumeric`](https://api.jquery.com/jQuery.isNumeric/) utility. Prefer `typeof`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.isNumeric();
```

âœ”ï¸ Examples of **correct** code:
```js
isNumeric();
myClass.isNumeric();
$div.isNumeric();
```

## Resources

* [Rule source](/src/rules/no-is-numeric.js)
* [Test source](/tests/rules/no-is-numeric.js)
