[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-is-window

Disallows the [`$.isWindow`](https://api.jquery.com/jQuery.isWindow/) utility.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.isWindow();
```

âœ”ï¸ Examples of **correct** code:
```js
isWindow();
myClass.isWindow();
```

## Resources

* [Rule source](/src/rules/no-is-window.js)
* [Test source](/tests/rules/no-is-window.js)
