[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-is-function

Disallows the [`$.isFunction`](https://api.jquery.com/jQuery.isFunction/) utility. Prefer `typeof`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.isFunction( expression( arg ) );

if ( $.isFunction( fn ) ) {
    g();
}
```

âœ”ï¸ Examples of **correct** code:
```js
isFunction();
myClass.isFunction();
$div.isFunction();
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
$.isFunction( expression( arg ) ); /* â†’ */ typeof expression( arg ) === 'function';

if ( $.isFunction( fn ) ) {        /* â†’ */ if ( typeof fn === 'function' ) {
    g();                           /* â†’ */     g();
}                                  /* â†’ */ }
```

## Resources

* [Rule source](/src/rules/no-is-function.js)
* [Test source](/tests/rules/no-is-function.js)
