[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-jquery-constructor

Disallows the jQuery constructor `$()`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' );
$( '#id' );
$( '.selector' );
$( '.selector', '.context' );
$( function () {} );
$( [] );
$( undefined );
$( false );
jQuery( 'div' );
```

‚úîÔ∏è Examples of **correct** code:
```js
foo.$( '.selector' );
foo.$( '<div>' );
foo.jQuery( '.selector' );
foo.jQuery( '<div>' );
$;
jQuery;
```

## Resources

* [Rule source](/src/rules/no-jquery-constructor.js)
* [Test source](/tests/rules/no-jquery-constructor.js)
