[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-map-collection

Disallows the [`.map`](https://api.jquery.com/map/) method. Prefer `Array#map`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).map();
$div.map();
$( 'div' ).first().map();
$( 'div' ).append( $( 'input' ).map() );
```

‚úîÔ∏è Examples of **correct** code:
```js
map();
[].map();
div.map();
div.map;
$.map();
$( 'select' ).val().map();
```

## Resources

* [Rule source](/src/rules/no-map-collection.js)
* [Test source](/tests/rules/no-map-collection.js)
