[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-node-name

Disallows the [`$.nodeName`](https://api.jquery.com/jQuery.nodeName/) utility.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.2`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.nodeName();
```

âœ”ï¸ Examples of **correct** code:
```js
nodeName();
myClass.nodeName();
$div.nodeName();
```

## Resources

* [Rule source](/src/rules/no-node-name.js)
* [Test source](/tests/rules/no-node-name.js)
