# no-now

Disallows the [`$.now`](https://api.jquery.com/jQuery.now/) utility. Prefer `(new Date).getTime()`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.now();
```

âœ”ï¸ Examples of **correct** code:
```js
now();
myClass.now();
$div.now();
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
$.now(); /* â†’ */ Date.now();
```

## Resources

* [Rule source](/src/rules/no-now.js)
* [Test source](/src/tests/no-now.js)
