# no-now

Disallows the [`$.now`](https://api.jquery.com/jQuery.now/) utility. Prefer `(new Date).getTime()`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.now();
```

âœ”ï¸ Examples of **correct** code:
```js
now();
myClass.now();
$div.now();
```

ğŸ”§ Examples of code **fixed** by using  `--fix`:
```js
$.now(); /* â†’ */ Date.now();
```

## Resources

* [Rule source](/src/rules/no-now.js)
* [Test source](/src/tests/no-now.js)
