[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-noop

Disallows the [`$.noop`](https://api.jquery.com/jQuery.noop/) property. Prefer `function(){}`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.noop;
$.noop();
```

âœ”ï¸ Examples of **correct** code:
```js
foo.noop;
foo.noop();
foo.noop( bar );
$.noOp;
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
$.noop;   /* â†’ */ ( function () {} );
$.noop(); /* â†’ */ ( function () {}() );
```

## Resources

* [Rule source](/src/rules/no-noop.js)
* [Test source](/tests/rules/no-noop.js)
