# no-on-ready

Disallows using the ready event on the document.

‚öôÔ∏è This rule is enabled in `plugin:no-jquery/deprecated-1.8`.

üîß The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( document ).on( 'ready', function () {} );
$document.on( 'ready', function () {} );
```

‚úîÔ∏è Examples of **correct** code:
```js
$( document ).on( 'click', function () {} );
$( document ).on();
$document.on( 'click', function () {} );
$document.on();
$document.on( ready );
$document.on( ready() );
$document.ready();
$( function () {} );
document.on( 'ready' );
document.on( 'ready', function () {} );
```

üîß Examples of code **fixed** by this rule:
```js
$( document ).on( 'ready', function () {} ); /* ‚Üí */ $( document ).ready( function () {} );
$document.on( 'ready', function () {} );     /* ‚Üí */ $document.ready( function () {} );
```

## Resources

* [Rule source](/src/rules/no-on-ready.js)
* [Test source](/tests/rules/no-on-ready.js)
