[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-other-methods

Disallows all methods not covered by more specific rules.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$div.add();
$div.height();
$div.prepend();
$div.otherMethod();
```

‚úîÔ∏è Examples of **correct** code:
```js
$div.each();
$div.find();
$div.map();
$div.otherProperty;
div.otherMethod();
$.otherUtil();
method();
```

## Resources

* [Rule source](/src/rules/no-other-methods.js)
* [Test source](/tests/rules/no-other-methods.js)
