[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-parse-html

Disallows the [`$.parseHTML`](https://api.jquery.com/jQuery.parseHTML/) utility. Prefer `DOMImplementation#createHTMLDocument`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.parseHTML( '<b>test</b>' );
```

âœ”ï¸ Examples of **correct** code:
```js
parseHTML( '<b>test</b>' );
'test'.parseHTML( '<b>test</b>' );
'<b>test</b>'.parseHTML;
```

## Resources

* [Rule source](/src/rules/no-parse-html.js)
* [Test source](/tests/rules/no-parse-html.js)
