# no-parse-xml

Disallows the [`$.parseXML`](https://api.jquery.com/jQuery.parseXML/) utility. Prefer `DOMParser#parseFromString`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/slim`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.parseXML( '<b>test</b>' );
```

âœ”ï¸ Examples of **correct** code:
```js
parseXML( '<b>test</b>' );
'test'.parseXML( '<b>test</b>' );
'<b>test</b>'.parseXML;
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
$.parseXML( '<b>test</b>' ); /* â†’ */ ( new window.DOMParser() ).parseFromString( '<b>test</b>', 'text/xml' );
```

## Resources

* [Rule source](/src/rules/no-parse-xml.js)
* [Test source](/tests/rules/no-parse-xml.js)
