# no-parse-json

Disallows the [`$.parseJSON`](https://api.jquery.com/jQuery.parseJSON/) utility. Prefer `JSON.parse`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/deprecated-3.0`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.parseJSON( '{}' );
```

âœ”ï¸ Examples of **correct** code:
```js
parseJSON();
$div.parseJSON();
'test'.parseJSON;
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
$.parseJSON( '{}' ); /* â†’ */ JSON.parse( '{}' );
```

## Resources

* [Rule source](/src/rules/no-parse-json.js)
* [Test source](/src/tests/no-parse-json.js)
