[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-parents

Disallows the [`.parents`](https://api.jquery.com/parents/) method. Prefer `Element#closest`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$( 'div' ).parents();
$div.parents();
$( 'div' ).first().parents();
$( 'div' ).append( $( 'input' ).parents() );
```

âœ”ï¸ Examples of **correct** code:
```js
parents();
[].parents();
div.parents();
div.parents;
```

## Resources

* [Rule source](/src/rules/no-parents.js)
* [Test source](/tests/rules/no-parents.js)
