[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-proxy

Disallows the [`$.proxy`](https://api.jquery.com/jQuery.proxy/) utility. Prefer `Function#bind`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.proxy( this.fn, context );
$.proxy( fn, context, arg1, arg2 );
$.proxy( context, 'fnName' );
$.proxy( context, 'fnName', arg1, arg2 );
```

âœ”ï¸ Examples of **correct** code:
```js
proxy( fn, context );
'test'.proxy( fn, context );
'test'.proxy;
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
$.proxy( this.fn, context );        /* â†’ */ this.fn.bind( context );
$.proxy( fn, context, arg1, arg2 ); /* â†’ */ fn.bind( context, arg1, arg2 );
```

## Resources

* [Rule source](/src/rules/no-proxy.js)
* [Test source](/tests/rules/no-proxy.js)
