[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-sub

Disallows the [`$.sub`](https://api.jquery.com/jQuery.sub/) utility.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-1.7`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.sub();
```

âœ”ï¸ Examples of **correct** code:
```js
sub();
'test'.sub();
'test'.sub;
```

## Resources

* [Rule source](/src/rules/no-sub.js)
* [Test source](/tests/rules/no-sub.js)
