[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-selector-prop

Disallows the [`.selector`](https://api.jquery.com/selector/) property.

üìã This rule is enabled in `plugin:no-jquery/deprecated-1.7`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).selector;
$div.selector;
$div.selector.prop;
$div.selector.method();
$( 'div' ).first().selector;
f( $div.selector );
$( 'div' ).append( $( 'input' ).selector );
```

‚úîÔ∏è Examples of **correct** code:
```js
selector;
div.selector;
$div.prop.selector;
$div.selector();
$div.selector( arg );
```

## Resources

* [Rule source](/src/rules/no-selector-prop.js)
* [Test source](/tests/rules/no-selector-prop.js)
