[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-text

Disallows the [`.text`](https://api.jquery.com/text/) method and `$.text` utility. Prefer `Node#textContent`.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$.text();
$( 'div' ).text();
$div.text();
$( 'div' ).first().text();
$( 'div' ).append( $( 'input' ).text() );
```

‚úîÔ∏è Examples of **correct** code:
```js
text();
[].text();
div.text();
div.text;
```

## Resources

* [Rule source](/src/rules/no-text.js)
* [Test source](/tests/rules/no-text.js)
