[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-type

Disallows the [`$.type`](https://api.jquery.com/jQuery.type/) utility. Prefer `typeof`/`instanceof`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.3`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.type();
```

âœ”ï¸ Examples of **correct** code:
```js
type();
myMethod.type();
myMethod.type;
```

## Resources

* [Rule source](/src/rules/no-type.js)
* [Test source](/tests/rules/no-type.js)
