[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-trim

Disallows the [`$.trim`](https://api.jquery.com/jQuery.trim/) utility. Prefer `String#trim`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/deprecated-3.5`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$.trim( ' test ' );
```

âœ”ï¸ Examples of **correct** code:
```js
trim( ' test ' );
' test '.trim();
' test '.trim;
$( 'input' ).text().trim();
$( 'input' ).data( 'foo' ).trim();
```

## Resources

* [Rule source](/src/rules/no-trim.js)
* [Test source](/tests/rules/no-trim.js)
