[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-visibility

Disallows the [`.show`](https://api.jquery.com/show/)/[`.hide`](https://api.jquery.com/hide/)/[`.toggle`](https://api.jquery.com/toggle/) methods.

üìã This rule is enabled in `plugin:no-jquery/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).show();
$div.show();
$( 'div' ).first().show();
$( 'div' ).append( $( 'input' ).show() );
$( 'div' ).hide();
$div.hide();
$( 'div' ).first().hide();
$( 'div' ).append( $( 'input' ).hide() );
$( 'div' ).toggle();
$div.toggle();
$( 'div' ).first().toggle();
$( 'div' ).append( $( 'input' ).toggle() );
```

‚úîÔ∏è Examples of **correct** code:
```js
show();
[].show();
div.show();
div.show;
hide();
[].hide();
div.hide();
div.hide;
toggle();
[].toggle();
div.toggle();
div.toggle;
```

## Resources

* [Rule source](/src/rules/no-visibility.js)
* [Test source](/tests/rules/no-visibility.js)
