[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-val

Disallows the [`.val`](https://api.jquery.com/val/) method. Prefer `HTMLInputElement#value`.

ğŸ“‹ This rule is enabled in `plugin:no-jquery/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$( 'div' ).val();
$div.val();
$( 'div' ).first().val();
$( 'div' ).append( $( 'input' ).val() );
```

âœ”ï¸ Examples of **correct** code:
```js
val();
[].val();
div.val();
div.val;
```

## Resources

* [Rule source](/src/rules/no-val.js)
* [Test source](/tests/rules/no-val.js)
