# no-load-shorthand

Disallows the [`.load`](https://api.jquery.com/load-event/) method when used as a shorthand for `.on( 'load', function )` or `.trigger( 'load' )`. Prefer `.on` or `.trigger`.

‚öôÔ∏è This rule is enabled in `plugin:no-jquery/deprecated-1.8`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).load( function () {} );
$div.load( function () {} );
$div.load();
$div.load( () => {} );
$( 'div' ).first().load( function () {} );
$( 'div' ).append( $( 'input' ).load( function () {} ) );
```

‚úîÔ∏è Examples of **correct** code:
```js
load();
[].load();
div.load();
div.load;
$.load();
$div.load( 'url' );
$div.load( couldBeUrl );
```

üîß Examples of code **fixed** by using  `--fix`:
```js
$( 'div' ).load( function () {} ); /* ‚Üí */ $( 'div' ).on( 'load', function () {} );
$div.load();                       /* ‚Üí */ $div.trigger( 'load' );
```

## Resources

* [Rule source](/src/rules/no-load-shorthand.js)
* [Test source](/src/tests/no-load-shorthand.js)
