# no-load-shorthand

Disallows the [`.load`](https://api.jquery.com/load-event/) method when used as a shorthand for `.on( 'load', function )` or `.trigger( 'load' )`. Prefer `.on` or `.trigger`.

âš™ï¸ This rule is enabled in `plugin:no-jquery/deprecated-1.8`.

## Rule details

âŒ Examples of **incorrect** code:
```js
$( 'div' ).load( function () {} );
$div.load( function () {} );
$div.load();
$div.load( () => {} );
$( 'div' ).first().load( function () {} );
$( 'div' ).append( $( 'input' ).load( function () {} ) );
```

âœ”ï¸ Examples of **correct** code:
```js
load();
[].load();
div.load();
div.load;
$.load();
$div.load( 'url' );
$div.load( couldBeUrl );
```

ðŸ”§ Examples of code **fixed** by using  `--fix`:
```js
$( 'div' ).load( function () {} );                        /* â†’ */ $( 'div' ).on( 'load', function () {} );
$div.load( function () {} );                              /* â†’ */ $div.on( 'load', function () {} );
$div.load();                                              /* â†’ */ $div.trigger( 'load' );
$div.load( () => {} );                                    /* â†’ */ $div.on( 'load', () => {} );
$( 'div' ).first().load( function () {} );                /* â†’ */ $( 'div' ).first().on( 'load', function () {} );
$( 'div' ).append( $( 'input' ).load( function () {} ) ); /* â†’ */ $( 'div' ).append( $( 'input' ).on( 'load', function () {} ) );
```

## Resources

* [Rule source](/src/rules/no-load-shorthand.js)
* [Test source](/src/tests/no-load-shorthand.js)
